sdk/attestation/azure-security-attestation/azure/security/attestation/_administration_client.py

```diff
@@ -130,37 +248,64 @@ def set_policy(self, attestation_type, attestation_policy, signing_key=None, **k
 
         """
 
-        policy_token = AttestationToken[GeneratedStoredAttestationPolicy](
+        # If the caller provided a signing key and certificate, validate that,
+        # otherwise use the default values from the service.
+        signing_key = kwargs.pop('signing_key', None)
+        signing_certificate = kwargs.pop('signing_certificate', None)
+        if not signing_key:
+            signing_key = self._signing_key
+        if not signing_certificate:
+            signing_certificate = self._signing_certificate
```

Comment:
Technically, it is possible, but in practice, I wouldn't recommend it - these really are tuples and while you technically can disconnect the two, it's not wise.

More importantly, there was a C&P issue that lost the call to validate_signing_keys.

Reply:
I was wrong, btw - the C&P issue was handled elsewhere. Since this confused me, I added a comment to explain a bit of what's oging on here. Bottom line is that functionally it's impossible to treat them as independent constructs. The class ctor saves self._signing_key and self._signing_certificate as their underlying cryptography types, so you have to provide both of the types to the ctor.